### DISCUSSION TITLE
Why are desktop notifications disabled when not using TLS?

### DISCUSSION THREAD

**Discussion by Fabian-G** [navidrome/navidrome#3427] [Q&A]
I don't publicly expose my navidrome instance and access it only through a VPN, so I don't see the point in using TLS.
I'm wondering why the desktop notification feature requires TLS. Is this some limitation of the browser API or are there any security implications I don't see?

**Comment by metalheim** [comment]
Browsers limit some APIs if the website is not using a "secure context".
See here for an explaination:
https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts

Basically, a secure context requires TLS connection or access from "localhost"
